src/main/java/org/microbean/microprofile/config/ConfigBuilder.java

```diff
@@ -93,30 +93,31 @@ public final Config build() {
     }
 
     ClassLoader classLoader = null;
-    
+
     if (this.addDiscoveredSources) {
       classLoader = this.classLoader;
       if (sources == null) {
         sources = new LinkedList<>();
       }
       sources.addAll(org.microbean.microprofile.config.Config.getDiscoveredConfigSources(classLoader));
     }
-    
-    if (sources != null) {
-      synchronized (this.sources) {
-        sources.addAll(this.sources);
-        Collections.sort(sources, ConfigSourceComparator.INSTANCE);
-      }
+
+    if (sources == null) {
+      sources = new LinkedList<>();
```

Comment:
This means that no matter what, `sources` will always have a non-null value by the end of this method. It might be simpler then to just initialize it to a `LinkedList` in the first place and avoid all these null checks.

Reply:
Perhaps also reserve sorting for the end then as well after all discovered, default, and user-supplied sources are in the list. If the list is empty, sorting should be instant.